driver.go

```diff
@@ -101,6 +102,12 @@ func parseURI(uri string) (*Options, error) {
 		}
 	}
 
+	clientId, ok := query["clientId"]
```

Comment:
could you add a sample for the usage? how the end user is going to set the user-agent suffix?

is that settable per query? or is that settable once the DBSQL go client is initialized?

Reply:
why is that called `clientId`? we should be consistent across DBSQL SDKs. what is it called in the other dbsql sdks. let's say pysql?

Reply:
Let's first align on that - are we fine with having `clientId` as part of DSN connection string? NodeJS SDK has:
```
 return client.connect({
        host,
        path: `/sql/1.0/endpoints/${endpointId}`,
        token,
        clientId: `my_client_id`,
    });
```
and Python SDK has:
```
databricks.sql.connect(**self.DUMMY_CONNECTION_ARGS, _user_agent_entry=”my_client_id")
```
Problem is that the current implementation of Golang SDK expects clients to supply a DSN connection string when connecting.

Reply:
Thanks @mattdeekay for quick check.

I see so NodeJs SDK is not consistent with PySQL. that's misfortunate :/

Could you please also check if JDBC/ODBC drivers also support this feature (adding user agent suffix) and if so what the config name is called in those Drivers?

Reply:
I'm happy to change "ClientId" to "UserAgentEntry" to align with Python SDK ("_user_agent_entry"), JDBC, and ODBC. 

However, if we want to move away from **explicit** DSN connection string to something like Python SDK's `databricks.sql.connect(**self.DUMMY_CONNECTION_ARGS, _user_agent_entry="my_client_id")`, that may need to be a separate effort that needs scoping and prioritization

Reply:
ok based on what you shared

JDBC -> UserAgentEntry
ODBC -> UserAgentEntry
Pysql -> _user_agent_entry
Nodejs -> clientId

Seems majority are using UserAgentEntry, python tried to make that pytonic user_agent_entry. It is Nodejs naming which diverged from the rest, we should fix this in NodeJs. 

We should try to be consistent with the existing SDKs. which in this case I would say is to use "UserAgentEntry". 

Because go already uses lower camel case e.g, `maxRows` in
```
databricks://:[your token]@[Workspace hostname][Endpoint HTTP Path]?timeout=1000&maxRows=1000
```

I would name this `userAgentEntry` to be consistent with the existing config names.